src/scheduler/server_info.c

```diff
@@ -704,21 +712,91 @@ query_server_dyn_res(server_info *sinfo)
 			/* Make sure file does not have open permissions */
 			err = tmp_file_sec(filename, 0, 1, S_IWGRP|S_IWOTH, 1);
 			if (err != 0) {
-				log_eventf(PBSEVENT_SECURITY, PBS_EVENTCLASS_SERVER, LOG_ERR, "server_dyn_res", 
+				log_eventf(PBSEVENT_SECURITY, PBS_EVENTCLASS_SERVER, LOG_ERR, "server_dyn_res",
 					"error: %s file has a non-secure file access, setting resource %s to 0, errno: %d",
 					filename, res->name, err);
-				(void) set_resource(res, res_zero, RF_AVAIL);
+				set_resource(res, res_zero, RF_AVAIL);
+				continue;
+			}
+
+			if (pipe(pdes) < 0) {
```

Comment:
Any chance you can put the fork() and pipe() code in a separate function?  As an added benefit, you can get rid of the goto which we don't use often in the scheduler.

Reply:
I'd rather not, I'd have to keep track of too many variables (pipes, pid, errno).
I guess I can put everything in if blocks, I'll try that.